ipaserver/install/cainstance.py

```diff
@@ -1396,11 +1396,18 @@ def is_crlgen_enabled(self):
         try:
             rewriteRuleDisabled = True
             p = re.compile(self.crl_rewrite_pattern)
-            with open(paths.HTTPD_IPA_PKI_PROXY_CONF) as f:
-                for line in f.readlines():
-                    if p.search(line):
-                        rewriteRuleDisabled = False
-                        break
+            if os.path.isfile(paths.HTTPD_IPA_PKI_PROXY_CONF):
+                with open(paths.HTTPD_IPA_PKI_PROXY_CONF) as f:
+                    for line in f.readlines():
+                        if p.search(line):
+                            rewriteRuleDisabled = False
+                            break
+            else:
+                # Installation seems to be in inconsistent state.
+                # Throwing an exception could e.g. inhibit uninstall
+                # (preventing cleanup), so let's not do that.
+                rewriteRuleDisabled = False
```

Comment:
If the code throws an InconsistentCRLGenConfigException, uninstall will catch the exception and proceed either with a prompt "are you sure, deleting your CRL gen master" or abort uninstall in non-interactive mode, unless --ignore-last-of-role is provided.

Reply:
It was throwing `IOError` due to missing file, causing whole `ipa-server-install` program to terminate.

Reply:
Yes, but my point is that the code could be modified by replacing the `raise RuntimeError` with a `raise InconsistentCRLGenConfigException` after `except IOError` in line 1405.

Reply:
Ah I see, thanks for clarifying.  I'll apply your suggestion, thanks @flo-renaud .